matter_server/client/connection.py

```diff
@@ -75,31 +75,23 @@ async def connect(self) -> None:
         info = cast(ServerInfoMessage, await self.receive_message_or_raise())
         self.server_info = info
 
-        # sdk version must match exactly
-        if info.sdk_version != chip_clusters_version():
-            await self._ws_client.close()
-            raise InvalidServerVersion(
-                f"Matter Server SDK version is incompatible: {info.sdk_version} "
-                f"version on the client is {chip_clusters_version()} "
-            )
-
         # basic check for server schema version compatibility
-        if info.schema_version < MIN_SCHEMA_VERSION:
-            # server version is too low, raise exception
+        if info.min_supported_schema_version > SCHEMA_VERSION:
+            # our schema version is too low and can't be handled by the server anymore.
             await self._ws_client.close()
             raise InvalidServerVersion(
-                f"Matter Server schema version is incompatible: {info.schema_version} "
-                f"a version is required that supports at least api schema {MIN_SCHEMA_VERSION} "
-                " - update the Matter Server to a more recent version."
+                f"Matter schema version is incompatible: {info.schema_version}, "
+                f"the server requires at least {info.min_supported_schema_version} "
+                " - update the Matter client to a more recent version or downgrade the server."
             )
-        if self.server_info.schema_version > MIN_SCHEMA_VERSION:
-            # server version is higher than expected, log only
+        if abs(self.server_info.schema_version - SCHEMA_VERSION) > 1:
```

Comment:
I think we can just drop this check? We’re either supported or not?

Reply:
yeah, I was trying to be too save. Removed it as it does not add any value